exercises/practice/space-age/.meta/example.f90

```diff
@@ -0,0 +1,60 @@
+
+module space_age
+  implicit none
+contains
+
+  double precision function age_in_years(planet, seconds)
+    !Mercury: orbital period 0.2408467 Earth years
+    !Venus: orbital period 0.61519726 Earth years
+    !Earth: orbital period 1.0 Earth years, 365.25 Earth days, or 31557600 seconds
+    !Mars: orbital period 1.8808158 Earth years
+    !Jupiter: orbital period 11.862615 Earth years
+    !Saturn: orbital period 29.447498 Earth years
+    !Uranus: orbital period 84.016846 Earth years
+    !Neptune: orbital period 164.79132 Earth years
+
+    character(len=*) :: planet
+    double precision :: seconds
+    
+    age_in_years = 0.d0
+    
+    select case (planet)
+    case ("Mercury")
+      !Mercury: orbital period 0.2408467 Earth years
+      age_in_years = seconds/31557600.d0/0.2408467d0
+
+    case ("Venus")
+      !Venus: orbital period 0.61519726 Earth years
+      age_in_years = seconds/31557600.d0/0.61519726d0
+
+    case ("Earth")
+      !Earth: orbital period 1.0 Earth years, 365.25 Earth days, or 31557600 seconds
+      age_in_years = seconds/31557600.d0
+
+    case ("Mars")
+      !Mars: orbital period 1.8808158 Earth years
+      age_in_years = seconds/31557600.d0/1.8808158d0
+
+    case ("Jupiter")
+      !Jupiter: orbital period 11.862615 Earth years
+      age_in_years = seconds/31557600.d0/11.862615d0
+
+    case ("Saturn")
+      !Saturn: orbital period 29.447498 Earth years
+      age_in_years = seconds/31557600.d0/29.447498d0
+
+    case ("Uranus")
+      !Uranus: orbital period 84.016846 Earth years
+      age_in_years = seconds/31557600.d0/84.016846d0
+
+    case ("Neptune")
+      !Neptune: orbital period 164.79132 Earth years
+      age_in_years = seconds/31557600.d0/164.79132d0
+
+    case default
+      write(*,*) 'ERROR: unknown planet: '//trim(planet)
+    end select 
+
+  end function
+
+end module
```

Comment:
Let me suggest a couple things:

1. Use constants, and put them in their own module
2. Only calculate the `age_in_years` once - we can use the `select` statement just to identify a multiplier

Also I'm not sure it's widespread, but do you have opinions on `intent(in)` and `intent(out)` etc? I'd think it's a best practice to specify these, but not sure if that's idiomatic or only what I've encountered in my limited experience

Let me know what you think

```suggestion

module orbital_constants
  implicit none
  
  !Earth: orbital period 1.0 Earth years, 365.25 Earth days, or 31557600 seconds
  double precision, parameter, private :: earth_year_seconds = 0.2408467d0
  
  ! A planet's orbit as measured in earth years
  double precision, parameter, private :: mercury_earth_years = 0.2408467d0
  double precision, parameter, private :: venus_earth_years = 0.61519726d0
  double precision, parameter, private :: mars_earth_years = 1.8808158d0
  double precision, parameter, private :: jupiter_earth_years = 11.862615d0
  double precision, parameter, private :: saturn_earth_years = 29.447498d0
  double precision, parameter, private :: uranus_earth_years = 84.016846d0
  double precision, parameter, private :: neptune_earth_years = 164.79132d0

end module orbital_constants

module space_age
  use orbital_constants
  implicit none
contains

  double precision function age_in_years(planet, seconds)
    character(len=*), intent(in) :: planet
    double precision, intent(in) :: seconds
    double precision, intent(out) :: age_in_years = 0.d0
    double precision :: n_earth_years
    
    select case (planet)
    case ("Mercury")
      n_earth_years = mercury_earth_years
    case ("Venus")
      n_earth_years = venus_earth_years
    case ("Earth")
      n_earth_years = 1.0d0
    case ("Mars")
      n_earth_years = mars_earth_years
    case ("Jupiter")
      n_earth_years = jupiter_earth_years
    case ("Saturn")
      n_earth_years = saturn_earth_years
    case ("Uranus")
      n_earth_years = uranus_earth_years
    case ("Neptune")
      n_earth_years = neptune_earth_years
    case default
      write(*,*) 'ERROR: unknown planet: '//trim(planet)
    end select
    
    age_in_years = seconds / earth_year_seconds / n_earth_years

  end function

end module
```

Reply:
You are totally right that your implementation is much nicer than mine. I'll build and test later